### DISCUSSION TITLE
Current method for installing additional NPM packages (Lightbox)

### DISCUSSION THREAD

**Discussion by jollydodo** [thuliteio/doks#1142] [Q&A] [answered]
Hey, I'm wondering what currently is the best way to install additional npm packages. There seems to have been somewhat of a shift in how doks is structured in the last years making some of the old posts irrelevant. Basically I'm trying to install a Lightbox so that users can zoom in on images. 

https://dimboxjs.com
seems to be a fairly new and updated light js package without the need for jQuery. 

Installing this the old way by just adding an inline javascript to my header works, but I want to make use the npm setup from Doks. So I ran "npm install dim box"  and tried to follow the guide from docsearch (https://getdoks.org/docs/add-ons/docsearch/) to get everything installed since this is basically trying to do the same thing.

meaning: 
1. import dimbox from 'dimbox'; added to assets/js/dimbox.js (also tried adding this to app.js)

2. changed my params file to add dimbox and then added this to the script-footer-custom.html
{{ if site.Params.add_ons.dimbox-}}
  {{ partial "footer/esbuild" (dict "src" "js/dimbox.js" "load" "async" "transpile" false) -}}
{{ end -}}

3. imported css files in app.css
@import "dimbox/dist/css/dimbox.min";

While the css imports fine, the website on the other hand does not want to run the dimbox script using this method. I'm having trouble debugging this, so I'm wondering what the correct approach is for such a npm package.

**Comment by h-enk** [comment, marked as answer]
The documentation will be updated — soonish 😄 

Since the DimBox script is relatively small, you could/should go this route (minimizing the number of requests): https://github.com/orgs/gethyas/discussions/442#discussioncomment-7828474

For larger scripts (like e.g. DocSearch or Bootstrap) the route you followed above is recommended.

1. I think the script is loaded faster than the page is ready. So, make it [`defer`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#defer):

    ```
    {{ partial "footer/esbuild" (dict "src" "js/dimbox.js" "load" "defer" "transpile" false) -}}
    ```

2. I would also load the SCSS (in stead of the CSS) — in `assets/scss/common/_custom.scss` (providing you more flexibility):

    ```scss
    // Put your custom SCSS code here
    
    // Set you own values for DimBox variables if needed
    $overlay-color-dark: #333;
    $overlay-color-light: #ccc;
    $overlay-opacity: .9;
    
    // Import DimBox SCSS
    @import 'dimbox/src/scss/dimbox';
    ```

3. Note that, for making it work in production you would probably have to update the Content Security Policy settings in `netlify.toml` (if you're hosting on Netlify) and update the PurgeCSS settings in `config/postcss.config.js` — add `'./node_modules/dimbox/src/scss/dimbox.scss',` to the ` ...whitelister` section.

**Reply by jollydodo** [reply]
Thank you! Worked like a charm. I also totally missed the above question on the hyas discussions board... I played around with it and decided to implement the fslightbox package. I indeed still get some problems with csp (also with the default inline-svg shortcode btw) but that is a worry for a later day.
